### ISSUE TITLE
'dotnet sln add' failed after upgrade to SDK 8.0

### ISSUE THREAD

**Issue by dominikjeske** [dotnet/sdk#36905]
### Describe the bug
When using new SDK 8.0 I cannot add projects having same name to one solution when having same root folder (for example .\X\A\Test.csproj and .\X\B\Test.csproj)
Prior SDK 8.0 there was no problem with this.
Exception message: `shark_all.sln already contains project Sample.App in solution folder areasSamples`

### To Reproduce

```
dotnet new sln --name all
md A
md B
cd A
dotnet new classlib --name "Test"
cd..
cd B
dotnet new classlib --name "Test"
cd..
dotnet sln add ".\A\Test\Test.csproj"
dotnet sln add ".\B\Test\Test.csproj"
// this works but when in same root
mv ".\A" ".\X\"
mv ".\B" ".\X\"
dotnet sln add ".\A\Test\Test.csproj"
dotnet sln add ".\B\Test\Test.csproj" -> already contains project Test in solution folder X 
```

### Further technical details
❯ dotnet --info
.NET SDK:
 Version:           8.0.100
 Commit:            57efcf1350
 Workload version:  8.0.100-manifests.71b9f198

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19045
 OS Platform: Windows
 RID:         win-x64
 Base Path:   C:\Program Files\dotnet\sdk\8.0.100\

.NET workloads installed:
 Workload version: 8.0.100-manifests.71b9f198
There are no installed workloads to display.

Host:
  Version:      8.0.0
  Architecture: x64
  Commit:       5535e31a71

.NET SDKs installed:
  6.0.415 [C:\Program Files\dotnet\sdk]
  7.0.112 [C:\Program Files\dotnet\sdk]
  7.0.203 [C:\Program Files\dotnet\sdk]
  8.0.100 [C:\Program Files\dotnet\sdk]

**Comment by rilehudd** [comment]
I've also run into this issue. My folders were a bit more nested--but seems like it is the same issue.

In my case the paths were:
A\B\Windows\ProjectA\ProjectA.csproj
A\B\Linux\ProjectA\ProjectA.csproj

**Comment by dominikjeske** [comment]
Any news for this issue? It is blocking us from update to SDK 8 because we heavily using 'dotnet sln add' in our scripts

**Comment by ADD-David-Antolin** [comment]
Same problem here.

**Comment by akoeplinger** [comment]
@marcpopMSFT @nagilson looks like this was caused by https://github.com/dotnet/sdk/pull/11846

**Comment by baronfel** [comment]
We think this should be fixed by https://github.com/dotnet/sdk/issues/36905

**Comment by dominikjeske** [comment]
When this issue will be resolved? I have 8.0.101 but still error exists

**Comment by akoeplinger** [comment]
The fix was merged after 8.0.101 so it should ship in the next release.

**Comment by marcpopMSFT** [comment]
The fix will be in the Feb 8.0.102 release.
